### PR TITLE
feat(repo): enable agents

### PR DESCRIPTION
**Pull request by vsavkin**
## Related Issue(s)

Fixes #

### CODE REVIEW

**Review by mandarini — APPROVED** [review]

File: .nx/workflows/agents.yaml [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+  - name: Git Clone
+    script: |
+      git init .
+      git remote add origin $GIT_REPOSITORY_URL
+      git fetch --no-tags --prune --progress --no-recurse-submodules --depth=1 origin +{{nxCommitSha}}:{{nxCommitRef}}
+      git checkout --progress --force -B {{nxBranch}} {{nxCommitRef}}
+
+  - name: Restore cache
+    script: |
+      nxw cache restore {{nxBranch}}-node_modules node_modules
```
Comment (meeroslav) [inline .nx/workflows/agents.yaml:26]: How does this work between the commits? Will every commit on the same branch use the same cached artifact despite potential dependency changes?

File: .nx/workflows/agents.yaml [hunk 2]
```diff
@@ -0,0 +1,64 @@
...
+
+  - name: Print Debug Info
+    script: |
+      echo $NX_CI_EXECUTION_ID
+      echo $NX_CI_EXECUTION_ENV
+
+  - name: Run Agent
+    script: |
+      source "$HOME/.cargo/env"
+      echo 123
```
Comment (meeroslav) [inline .nx/workflows/agents.yaml:57]: This should be removed

File: docs/generated/cli/affected-graph.md [hunk 3]
```diff
@@ -1,168 +0,0 @@
----
```
Comment (meeroslav) [inline docs/generated/cli/affected-graph.md:1]: These generated files should not be removed

File: .circleci/config.yml [hunk 4]
```diff
@@ -164,42 +83,35 @@ jobs:
...
-          name: Check Documentation
-          command: pnpm nx documentation --no-dte
-          no_output_timeout: 20m
       - run:
           name: Run Checks/Lint/Test/Build
           no_output_timeout: 60m
           command: |
             pids=()
-
-            pnpm nx-cloud record -- nx format:check --base=$NX_BASE --head=$NX_HEAD &
```
Comment (meeroslav) [inline .circleci/config.yml:187]: Why are we removing these?

**Review by meeroslav — CHANGES_REQUESTED** [review]
Left some notes

**Pull request merged**